### PR TITLE
ref: Retire Hierarchical Grouping

### PR DESCRIPTION
**Pull request by Swatinem**
This has the consequence that existing users of the mobile grouping strategy (aka hierarchical grouping) will be automatically migrated to the default grouping config via automatic grouping updates.

The change to `project_grouping_configs` should stop advertising the strategy as "latest".

### CODE REVIEW

File: src/sentry/projectoptions/defaults.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # epoch instead.
 LEGACY_GROUPING_CONFIG = "legacy:2019-03-12"
 DEFAULT_GROUPING_CONFIG = "newstyle:2019-10-29"
-BETA_GROUPING_CONFIG = "mobile:2021-02-12"
+BETA_GROUPING_CONFIG = "beta:2022-12-20"
```
Comment (mitsuhiko) [inline src/sentry/projectoptions/defaults.py:17]: I would already name this `newstyle:2022-12-20` so we can just make it stable with that name when we release it.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]
This makes sense to me but i think we should separate the retirement from the new grouping strategy. Otherwise we have to roll them out simultaneously.  I believe the date for hierarchical retirement is January?

**Comment by Swatinem** [comment]
> This makes sense to me but i think we should separate the retirement from the new grouping strategy.

👍🏻 changes this PR so it only includes removal of the setting, thus triggering auto-upgrades to the latest supported version.

**Comment by mitsuhiko** [comment]
Is it alive

**Comment by Swatinem** [comment]
I believe the email that was sent out mentioned early march? so we still gotta wait until then?

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**